File: packages/live-update/android/src/main/java/io/capawesome/capacitorjs/plugins/liveupdate/classes/results/SyncResult.java
package io.capawesome.capacitorjs.plugins.liveupdate.classes.results;

import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.liveupdate.interfaces.Result;
import org.json.JSONObject;

public class SyncResult implements Result {

  @Nullable
  private  final String currentBundleId;
    @Nullable
    private final String nextBundleId;

    public SyncResult(@Nullable String currentBundleId, @Nullable String nextBundleId) {
        this.currentBundleId = currentBundleId;
      this.nextBundleId = nextBundleId;
    }

    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("currentBundleId", currentBundleId == null ? JSONObject.NULL : currentBundleId);
        result.put("nextBundleId", nextBundleId == null ? JSONObject.NULL : nextBundleId);
        return result;
    }
}

File: packages/live-update/ios/Plugin/Classes/Results/SyncResult.swift
import Foundation
import Capacitor

@objc public class SyncResult: NSObject, Result {
    let nextBundleId: String?

    init(nextBundleId: String?) {
        self.nextBundleId = nextBundleId
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["nextBundleId"] = nextBundleId == nil ? NSNull() : nextBundleId
        return result as AnyObject
    }
}
